src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java

```diff
@@ -116,11 +131,22 @@ public void processDelta(Message<ResourceChangedData> resourceChangedMessage) {
         logger.trace(String.format("Performing a PATCH with "
                 + "company number %s for contextId %s", companyNumber, logContext));
 
-        if (links == null) {
-            links = new Links();
+        if (links == null || links.getInsolvency() == null) {
+            final ApiResponse<CompanyInsolvency> companyInsolvencResponse = companyInsolvencyService
+                    .getCompanyInsolvency(logContext, companyNumber);
+
+            HttpStatus statusCode = HttpStatus.valueOf(companyInsolvencResponse.getStatusCode());
+
+            if (statusCode.is2xxSuccessful()) {
+                links = new Links();
+                links.setInsolvency(String.format("/company/%s/insolvency", companyNumber));
+            } else {
+                String message = "Response from companyInsolvencyService";
+                logger.errorContext(logContext, message, null, logMap);
```

Comment:
Probably can have more appropriate message like main delta update not yet completed, will re-try etc

Reply:
Relevant code change done